appinventor/blocklyeditor/src/blocklyeditor.js

```diff
@@ -40,6 +40,31 @@ Blockly.configForTypeBlock = {
 Blockly.BlocklyEditor.render = function() {
 };
 
+unboundVariableHandler = function(myBlock, yailText) {
+  var unbound_vars = Blockly.LexicalVariable.freeVariables(myBlock);
+  unbound_vars = unbound_vars.toList();
+  if (unbound_vars.length == 0) {
+    Blockly.ReplMgr.putYail(yailText, myBlock);
+  } else {
+    var form = "<form>Enter values for:<br>";
```

Comment:
Instead of hard-coding the text, add an entry to the msg/en/_messages.js file so that we can internationalize this message. Otherwise, everyone will see it in English even if it's not the active language in the editor.